### PR TITLE
Convert more views to use `ContentMenuMixin`

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/campaigns/views.py [hunk 1]
```diff
@@ -123,79 +123,47 @@ def form_valid(self, form):
...
-        def get_gear_links(self):
-            links = []
-
+        def build_content_menu(self, menu):
             if self.object.is_archived:
                 if self.has_org_perm("campaigns.campaign_activate"):
-                    links.append(
-                        dict(
-                            title="Activate",
-                            js_class="posterize activate-campaign",
```
Comment (rowanseymour) [inline temba/campaigns/views.py:140]: additional styles here not used

File: temba/campaigns/views.py [hunk 2]
```diff
@@ -123,79 +123,47 @@ def form_valid(self, form):
...
 
-            user = self.get_user()
-            if user.is_superuser or user.is_staff:
-                links.append(
-                    dict(
-                        title=_("Service"),
-                        posterize=True,
-                        href=f'{reverse("orgs.org_service")}?organization={self.object.org_id}&redirect_url={reverse("campaigns.campaign_read", args=[self.object.uuid])}',
-                    )
+            if self.request.user.is_staff:
```
Comment (rowanseymour) [inline temba/campaigns/views.py:161]: superuser is staff

File: temba/utils/views.py [hunk 3]
```diff
@@ -304,6 +307,21 @@ def add_modax(
...
         )
 
+    def as_items(self):
+        """
+        Reduce groups to a flat list of items separated by a divider.
+        """
+        items = []
+        for group in self.groups:
+            if not group:
+                continue
```
Comment (rowanseymour) [inline temba/utils/views.py:317]: this means you can call `menu.new_group()` inside `build_content_menu` without worrying about creating empty groups of menu items

File: templates/contacts/contact_list.haml [hunk 4]
```diff
@@ -200,9 +200,8 @@
                             {{current_group.name}}
                           -else
                             {{title}}
-                  -if view.get_gear_links
```
Comment (rowanseymour) [inline templates/contacts/contact_list.haml:203]: not needed and inefficient to call that method twice

**Review by norkans7 — APPROVED** [review]

**Pull request merged**